packages/hub/src/lib/ExportsDropdown.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { FC, useMemo, useState } from "react";
+import { PropsWithChildren } from "react";
+import { FormProvider, useFieldArray, useForm } from "react-hook-form";
+import { graphql, useFragment } from "react-relay";
+
+import {
+  ButtonWithDropdown,
+  CommentIcon,
+  Dropdown,
+  DropdownMenu,
+  DropdownMenuActionItem,
+  DropdownMenuHeader,
+  DropdownMenuModalActionItem,
+  LinkIcon,
+  ScaleIcon,
+  ShareIcon,
+  TextAreaFormField,
+  TextTooltip,
+} from "@quri/ui";
+import { modelExportRoute, modelForRelativeValuesExportRoute } from "@/routes";
+import { DropdownMenuNextLinkItem } from "@/components/ui/DropdownMenuNextLinkItem";
+
+type ModelExport = { variableName: string; title?: string };
+type RelativeValuesExport = { slug: string; variableName: string };
+
+const nonRelativeValuesExports = (
+  modelExports: ModelExport[],
+  relativeValuesExports: RelativeValuesExport[]
+) =>
+  modelExports.filter(
+    (exportItem) =>
+      !relativeValuesExports.find(
+        ({ variableName: v }) => v === exportItem.variableName
+      )
+  );
+
+//It's a bit awkward that this here, but it's fairly closely coupled to ExportsDropdown.
+export const totalImportLength = (
+  modelExports: ModelExport[],
+  relativeValuesExports: RelativeValuesExport[]
+) =>
+  nonRelativeValuesExports(modelExports, relativeValuesExports).length +
+  relativeValuesExports.length;
+
+export const ExportsDropdown: FC<
+  PropsWithChildren<{
+    modelExports: ModelExport[];
+    relativeValuesExports: RelativeValuesExport[];
```

Comment:
With recommended Relay patterns (which I think we should follow), you shouldn't have to type these explicitly, or do transformations to build these props.

Here's what you can do instead (I'm typing this in review comments, haven't checked if the code is valid):
1) write a custom fragment that asks for props that `ExportsDropdown` needs:

```typescript
const revision = useFragment(
  graphql`
    fragment ExportsDropdown on ModelRevision {
      exports {
        id
        variableName
        title
      }
      relativeValuesExports {
        id
        variableName
        definition {
          id
          slug
        }
      }
    }
  `, revisionRef
)`;
```

2) Take `revisionRef: ExportsDropdown$key` as a prop

3) Pass this fragment in `ModelLayout` and `ModelCard` queries, `...ExportsDropdown`

Then only `ExportsDropdown` component would know which props it needs; for example, if we add types on exported variables, you won't have to modify both `ModelCard` and `ModelLayout` to select this new field.

This leaves a problem of checking if `totalImportLength` is non-zero. (btw, why is it `totalImportLength` and not `totalExportLength`?)

Several ways to solve it:
1) with hooks:
- add `useModelHasExports` hook, that _also_ asks for these fields (maybe just for `id` on them because you don't need anything else)
- pass it as a fragment too, `...useModelHasExports`
- take fragment ref as a parameter and return boolean from a hook
See https://github.com/quantified-uncertainty/squiggle/blob/931a631ea3698087fe8366e2b638077ed1d922e4/packages/hub/src/app/groups/%5Bslug%5D/hooks.ts for an example of this pattern.

2) with an extra backend field; it's not hard to add an new `totalExportCount` field on the backend and then select it
(btw, we could filter out relative values exports from model exports on the backend too, maybe with a field argument, `exports(excludeRelativeValues: true) { ... }`

One more thing: when a field is a Relay node (or, in our case, a Prisma DB row, i.e. it's constructed with `builder.prismaNode`), you should always select its `id`, even if you don't need it. If you do it, Relay will store it in its normalized cache and update it properly on mutations automatically. It won't affect anything in the current code, but it's a good practice to avoid bugs with data consistency in the future.